cli/src/cmd/create.rs

```diff
@@ -96,7 +96,11 @@ impl CreateArgs {
         let factory = ContractFactory::new(abi, bin, Arc::new(provider));
 
         let deployer = factory.deploy_tokens(args)?;
-        let deployer = if self.legacy || is_legacy(chain) { deployer.legacy() } else { deployer };
+        let deployer = if self.legacy || Chain::try_from(chain).unwrap().is_legacy() {
```

Comment:
let's make `legacy == false` the default if the `chain` is an unknown chain
```suggestion
        let deployer = if self.legacy || Chain::try_from(chain).map(Chain::is_legacy).unwrap_or_default() {
```

Reply:
wait, I'm confused, isn't `chain` already of type `Chain`?

Reply:
hmm here it is a u64 since it is coming from the rpc provider